test/test_utils/__init__.py

```diff
@@ -862,6 +865,8 @@ def parse_canary_images(framework, region):
                 versions_counter[version]["tr"] = True
             elif "inf" in tag_str:
                 versions_counter[version]["inf"] = True
+            
+            pre_populated_py3_version[version] = match.group(3)
```

Comment:
so sometimes we might do a framework release with multiple python versions. If there are multiple python versions associated with a fw version, we may want to run canaries on all python versions. thus we can probably make this a list and iterate through while appending to the images list